### PR TITLE
added new reference books

### PR DESCRIPTION
**Pull request by cheta-nyadav**
## What does this PR do?
Add resources to  the list.

### Why is this valuable (or not)?
It can be really valuable for people getting started with programming.

### How do we know it's really free?
I have cross checked on my side, they're free!

### For book lists, is it a book? For course lists, is it a course? etc.
it is a book's pdf, ebook.

## Checklist:
- [ x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [ x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x ] Include author(s) and platform where appropriate.
- [x ] Put lists in alphabetical order, correct spacing.
- [x ] Add needed indications (PDF, access notes, under construction).
- [x ] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -538,6 +538,7 @@ Books on general-purpose programming that don't focus on a specific language are
 * [Joint Strike Fighter, C++ Coding Standards](http://www.stroustrup.com/JSF-AV-rules.pdf) - Bjarne Stroustrup (PDF)
 * [Learn C++ Programming Language](http://www.tutorialspoint.com/cplusplus/cpp_tutorial.pdf) - Tutorials Point (PDF)
 * [LearnCpp.com](https://www.learncpp.com) (HTML)
+* [The Complete Reference C++,Herbert Schildt](https://www.micc.unifi.it/bertini/download/labtecinf/C%2B%2B%20A%20Beginner's%20Guide%202nd%20Edition%20(2003).pdf) - Second Edition
```
Comment (rng70) [inline books/free-programming-books-langs.md:541]: move this line to line 550

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -1054,6 +1055,7 @@ Books on general-purpose programming that don't focus on a specific language are
 * [Effective Java, Third Edition](https://ia801009.us.archive.org/16/items/effectivejava2017addisonwesley/Effective%20Java%20%282017%2C%20Addison-Wesley%29.pdf) - Joshua Bloch (PDF) *(:card_file_box: archived)*
 * [Essential Java](https://www.programming-books.io/essential/java/) - Krzysztof Kowalczyk, StackOverflow Contributors
 * [Google's Java Style Guide](https://google.github.io/styleguide/javaguide.html)
+* [Java,The Complete Reference](https://srikarthiks.files.wordpress.com/2016/07/java-the-complete-reference-8th-edition.pdf) - 8th edition
```
Comment (rng70) [inline books/free-programming-books-langs.md:1058]: move this line to line 1072

**Review by rng70 — CHANGES_REQUESTED** [review]

File: books/free-programming-books-langs.md (hunk 1 above)
Comment (rng70) [inline books/free-programming-books-langs.md:541]: You must follow the rule described [here](https://github.com/vhf/remark-lint-books-links)
For example
```test
[Another Awesome Book](http://example.com/book.pdf) - John Doe (PDF)
```
So for your case, the proper way to do this is
```text
[The Complete Reference C++ (second edition)](https://www.micc.unifi.it/bertini/download/labtecinf/C%2B%2B%20A%20Beginner's%20Guide%202nd%20Edition%20(2003).pdf) - Herbert Schildt (PDF)
```

File: books/free-programming-books-langs.md (hunk 2 above)
Comment (rng70) [inline books/free-programming-books-langs.md:1058]: also, follow the same rule here

**Review by rng70 — CHANGES_REQUESTED** [review]

File: books/free-programming-books-langs.md (hunk 2 above)
Comment (cheta-nyadav) [reply books/free-programming-books-langs.md:1058]: ok, thank you so much.

**Comment by cheta-nyadav** [comment]
@rng70 please check now,

**Pull request closed without merging**